packages/ui/src/viewer-table/ConditionalLogicModal/ConditionalRestrictionDetails.tsx

```diff
@@ -0,0 +1,200 @@
+	const computeRestrictions = [
+		{
+			condition: rangeMatch.min !== undefined,
+			prefix: 'is at least',
+			content: `${rangeMatch.min}`,
+		},
+		{
+			condition: rangeMatch.exclusiveMin !== undefined,
+			prefix: 'is greater than',
+			content: `${rangeMatch.exclusiveMin}`,
+		},
+		{
+			condition: rangeMatch.max !== undefined,
+			prefix: 'is at most',
+			content: `${rangeMatch.max}`,
+		},
+		{
+			condition: rangeMatch.exclusiveMax !== undefined,
+			prefix: 'is less than',
+			content: `${rangeMatch.exclusiveMax}`,
+		},
+	];
+
+	const computedRestrictionItems = computeRestrictions.filter((item) => item.condition);
+	return computedRestrictionItems ?
+			<Fragment>
+				{computedRestrictionItems
+					.map((computedRestrictionItem) => `${computedRestrictionItem.prefix} ${computedRestrictionItem.content}`)
+					.join(' and ')}
+			</Fragment>
+		:	undefined;
+};
+
+const existsMatch = (existsMatch: MatchRuleExists) => {
+	return <Fragment>{existsMatch ? 'has a value' : 'is empty'}</Fragment>;
+};
+
+const countMatch = (countMatch: MatchRuleCount) => {
+	if (typeof countMatch === 'number') {
+		return (
+			<Fragment>
+				has exactly {countMatch} item{countMatch === 1 ? '' : 's'}
+			</Fragment>
+		);
+	}
+
+	const computeRestrictions = [
+		{
+			condition: countMatch.min !== undefined,
+			prefix: 'has',
+			content: `${countMatch.min} or more items`,
+		},
+		{
+			condition: countMatch.exclusiveMin !== undefined,
+			prefix: 'has',
+			content: `more than ${countMatch.exclusiveMin} items`,
+		},
+		{
+			condition: countMatch.max !== undefined,
+			prefix: 'has',
+			content: `${countMatch.max} or fewer items`,
+		},
+		{
+			condition: countMatch.exclusiveMax !== undefined,
+			prefix: 'has',
+			content: `fewer than ${countMatch.exclusiveMax} items`,
+		},
+	];
+
+	const computedRestrictionItems = computeRestrictions.filter((item) => item.condition);
+
+	return computedRestrictionItems.length > 0 ?
+			<Fragment>{computedRestrictionItems.map((item) => `${item.prefix} ${item.content}`).join(' and ')}</Fragment>
+		:	undefined;
+};
+
+/**
+ * Renders conditional restriction details as formatted text based on the provided conditions and match case.
+ * @param conditions Array of restriction conditions to be rendered.
+ * @param matchCase Specifies how the conditions are joined (all, any, none).
+ * @returns JSX elements containing the formatted conditional restriction details.
+ */
+
+export const ConditionalRestrictionDetails = (conditions: RestrictionCondition[], matchCase: ArrayTestCase) => {
+	const conjunctionText = getConjunctionText(matchCase);
+
+	return (
+		<Fragment>
+			{conditions.map((condition, index) => {
+				const { fields, match } = condition;
+				return (
+					<Fragment key={index}>
+						{condition.case && renderFields(fields, condition.case)}
+						{match.value !== undefined && condition.case !== undefined && (
+							<Fragment> {valueMatch(match.value, condition.case, fields.length)}</Fragment>
```

Comment:
All these matches probably do not need the nested Fragments, you can put strings/jsx nodes here without them.